src/hotspot/share/opto/subnode.cpp

```diff
@@ -1529,10 +1529,46 @@ Node *BoolNode::Ideal(PhaseGVN *phase, bool can_reshape) {
     }
   }
 
+  const int cmp1_op = cmp1->Opcode();
+  const int cmp2_op = cmp2->Opcode();
+
+  // Change x +- Integer.MIN_VALUE <=> y +- Integer.MIN_VALUE into x u<=> y
```

Comment:
The comment should include `cmp2_op == Op_ConI` case.
Also it is not clear from comment and code  if different operations are allowed on both side: `x - MIN_INT <= y + MIN_INT`